### PR TITLE
Feature/sec 17 1

### PR DESCRIPTION
**Pull request by gizet**

### CODE REVIEW

File: src/main/java/ro/secur/profile/configuration/SecurityConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+
+        http
+                .csrf().disable()
+                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
+                .and()
+                .authorizeRequests()
+                .antMatchers(LOGIN_API, MY_PROFILE_API).permitAll()
```
Comment (CosminaIacob) [inline src/main/java/ro/secur/profile/configuration/SecurityConfiguration.java]: We should have access to my profile only if we are authenticated. Please add a TODO here to change this when the /myprofile login will be completely implemented

File: src/main/java/ro/secur/profile/service/LoginService.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        this.jwtConfig = jwtConfig;
+    }
+
+    /**
+     * @param request user and password
+     * @return Jwt from header "Authorization"
+     */
+    public String login(AuthenticationRequest request) {
+        String token = null;
+        HttpEntity<String> entity = createRequestEntity(request);
```
Comment (CosminaIacob) [inline src/main/java/ro/secur/profile/service/LoginService.java]: Wouldn't **request** or **requestEntity** be a more appropriate name?

File: src/main/java/ro/secur/profile/service/SkillService.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.web.client.RestTemplate;
+import ro.secur.profile.security.TokenStorage;
+
+
+@Service
+public class SkillService {
+
+    private static final String SECUR_SKILLS = "http://secur-skills/api/v1/skills";
+
+    private static final String TOKEN_KEY = "Bearer ";
```
Comment (CosminaIacob) [inline src/main/java/ro/secur/profile/service/SkillService.java]: This is the token prefix, not key.

File: src/main/java/ro/secur/profile/configuration/SecurityConfiguration.java (hunk 1 above)
Comment (gizet) [reply src/main/java/ro/secur/profile/configuration/SecurityConfiguration.java]: ok

File: src/main/java/ro/secur/profile/service/LoginService.java (hunk 2 above)
Comment (gizet) [reply src/main/java/ro/secur/profile/service/LoginService.java]: done

File: src/main/java/ro/secur/profile/service/SkillService.java (hunk 3 above)
Comment (gizet) [reply src/main/java/ro/secur/profile/service/SkillService.java]: done

File: src/main/java/ro/secur/profile/util/Api.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package ro.secur.profile.util;
+
+public class Api {
+
+    private static final String PREFIX_URL = "/api/v1";
```
Comment (CosminaIacob) [inline src/main/java/ro/secur/profile/util/Api.java]: do we need this class if we are using the path from app.properties?

File: src/main/resources/application-local.properties [hunk 5]
```diff
@@ -1 +1,11 @@
...
+
+#microservice name
+spring.application.name=secur-profile
+
+#filter
+application.jwt.secretkey=madkagdfgkjfhjrkojiurgoijdsgjkldfgdislfjspfo[fasdkfodkgdosmgbakw[pfkb
+application.jwt.tokenPrefix=Bearer 
+
+#basePath for all urls
+spring.mvc.servlet.path=/api/v1
```
Comment (CosminaIacob) [inline src/main/resources/application-local.properties]: where is this used?

File: src/main/java/ro/secur/profile/security/TokenStorage.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+import lombok.Getter;
+import lombok.Setter;
+
+/*
+* This class is temporary and is just for testing purposes
+ */
+public class TokenStorage {
+
+    private static String TOKEN;
```
Comment (ftansanu) [inline src/main/java/ro/secur/profile/security/TokenStorage.java]: If this variable is not final please use camel case naming. Please also change in this class where u have used TOKEN with uppercase

File: src/main/java/ro/secur/profile/security/TokenStorage.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
+package ro.secur.profile.security;
+
+import lombok.Getter;
+import lombok.Setter;
+
+/*
+* This class is temporary and is just for testing purposes
+ */
+public class TokenStorage {
```
Comment (ftansanu) [inline src/main/java/ro/secur/profile/security/TokenStorage.java]: use lombok

File: src/main/java/ro/secur/profile/service/SkillService.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    RestTemplate restTemplate;
+
+    public SkillService(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    public ResponseEntity<String> getSkillData() {
+
+        final HttpHeaders headers = new HttpHeaders();
+        headers.add("Authorization", TOKEN_PREFIX + TokenStorage.getTOKEN());
```
Comment (ftansanu) [inline src/main/java/ro/secur/profile/service/SkillService.java]: I think there is a constant in org.springframework.http.HttpHeaders class for this hardcoded string

File: src/main/java/ro/secur/profile/util/Api.java (hunk 4 above)
Comment (gizet) [reply src/main/java/ro/secur/profile/util/Api.java]: pentru moment da. le vom restructura la un urmator commit cand pot sa si testez exact ce se intampla acolo.

File: src/main/resources/application-local.properties (hunk 5 above)
Comment (gizet) [reply src/main/resources/application-local.properties]: in controllere

File: src/main/java/ro/secur/profile/security/TokenStorage.java (hunk 7 above)
Comment (gizet) [reply src/main/java/ro/secur/profile/security/TokenStorage.java]: nu functioneaza pentru variabile statice.

File: src/main/java/ro/secur/profile/security/TokenStorage.java (hunk 6 above)
Comment (gizet) [reply src/main/java/ro/secur/profile/security/TokenStorage.java]: changed

File: src/main/java/ro/secur/profile/security/TokenStorage.java (hunk 7 above)
Comment (ftansanu) [reply src/main/java/ro/secur/profile/security/TokenStorage.java]: da corect

File: src/main/java/ro/secur/profile/service/SkillService.java (hunk 8 above)
Comment (ftansanu) [reply src/main/java/ro/secur/profile/service/SkillService.java]: The above comment is not resolved.

**Pull request merged**